### ISSUE TITLE
Reading: Confidence Modeling for Neural Semantic Parsing

### ISSUE THREAD

**Issue by a1da4** [a1da4/paper#49]
## 0. Paper
[Confidence Modeling for Neural Semantic Parsing](https://www.aclweb.org/anthology/P18-1069/)

## 1. What is it?
They used **model, data, input uncertainty** into semantic parsing.

## 2. What is amazing compared to previous studies?
They categorized the causes of uncertainty into
* model
* data
* input

To estimate **what we do not know**.

### 3.1 Model Uncertainty
There are 3 metrics

#### 3.1.1 Dropout Pertubation
Use Monte Carlo Droput to estimate model uncertainty.

<img width="300" alt="スクリーンショット 2019-12-18 0 38 44" src="https://user-images.githubusercontent.com/45454055/71010147-c07f4580-212e-11ea-8e95-824bc216c857.png">

<img width="300" alt="スクリーンショット 2019-12-18 0 38 57" src="https://user-images.githubusercontent.com/45454055/71010171-c7a65380-212e-11ea-85ef-2ff6ba0552e2.png">

Calculate variance.
<img width="300" alt="スクリーンショット 2019-12-18 0 54 41" src="https://user-images.githubusercontent.com/45454055/71011564-fae9e200-2130-11ea-824a-19affbddca83.png">

#### 3.1.2 Gaussian Noise
Adding Gaussian noise, there are 2 methods.
Dropout was a Bernoulli distribution, but this noise is a Gaussian distribution.
<img width="300" alt="スクリーンショット 2019-12-18 1 01 06" src="https://user-images.githubusercontent.com/45454055/71012083-e1956580-2131-11ea-9167-6652c82eb8a6.png">
<img width="300" alt="スクリーンショット 2019-12-18 1 01 28" src="https://user-images.githubusercontent.com/45454055/71012121-ee19be00-2131-11ea-818c-39105a463679.png">

After that, calculating variance as a dropout.

#### 3.1.3 Posterior Probability
Use posterior probability.
<img width="300" alt="スクリーンショット 2019-12-18 1 04 56" src="https://user-images.githubusercontent.com/45454055/71012386-697b6f80-2132-11ea-8f1d-e510a086a8db.png">

#### 3.2.1 Probability of Input
They trained a Language Model on training data.
After that, calculate the probability of input.

#### 3.2.2 Number of Unknown Tokens
Use a number of unknown tokens in the **input**.

### 3.3 Input Uncertainty
To measure uncertainty cause by **ambiguous inputs**, they used 2 metrics.

#### 3.3.1 Variance of Top Candidates
Calculate top-k candidates variance.
<img width="164" alt="スクリーンショット 2019-12-18 1 18 40" src="https://user-images.githubusercontent.com/45454055/71013627-549fdb80-2134-11ea-90b5-d519906e536c.png">

#### 3.3.2 Entropy of Decoding
Moreover, they used the sequence level entropy.
a' is sampled by using Monte Carlo sampling.

<img width="314" alt="スクリーンショット 2019-12-18 1 20 53" src="https://user-images.githubusercontent.com/45454055/71014037-a3e60c00-2134-11ea-9d26-b9f0604683a9.png">

### 3.4 Confidence Scoreing
Using XGBoost regressor.
After that, wrapped by a logistic function.

## 4. How did validate it?
Task: semantic parsing
Data: Django, IFTTT
<img width="442" alt="スクリーンショット 2019-12-18 1 30 02" src="https://user-images.githubusercontent.com/45454055/71014839-eb20cc80-2135-11ea-8bb3-4f86554b44d9.png">

Table2: Model uncertainty is the best feature.

<img width="442" alt="スクリーンショット 2019-12-18 1 31 47" src="https://user-images.githubusercontent.com/45454055/71014979-29b68700-2136-11ea-9dc2-f92d5e61a51d.png">

Table5: Dropout and Gaussian noise are important for scoring.

<img width="285" alt="スクリーンショット 2019-12-18 1 33 05" src="https://user-images.githubusercontent.com/45454055/71015065-58346200-2136-11ea-8316-90a82fd407ca.png">

Figure3: The tradeoff between output proportion and F1score.

**Comment by a1da4** [comment]
#9 Improving Back-Translation with Uncertainty-based Confidence Estimation
